geode-core/src/test/java/org/apache/geode/management/internal/functions/RestoreRedundancyFunctionTest.java

```diff
@@ -0,0 +1,163 @@
+
+    verify(mockOperation).includeRegions(new HashSet<>(request.getIncludeRegions()));
+    verify(mockOperation).excludeRegions(new HashSet<>(request.getExcludeRegions()));
+    verify(mockOperation).shouldReassignPrimaries(request.getReassignPrimaries());
+  }
+
+  @Test
+  public void executeFunctionSetsIncludedAndExcludedRegionsOnRestoreRedundancyOperationWhenNull() {
+    function.execute(mockContext);
+
+    verify(mockOperation).includeRegions(null);
+    verify(mockOperation).excludeRegions(null);
+    verify(mockOperation).shouldReassignPrimaries(true);
+  }
+
+  @Test
+  public void executeFunctionUsesStatusMethodWhenIsStatusCommandIsTrue() {
+    when(mockOperation.redundancyStatus()).thenReturn(mockResults);
+    when(mockResults.getRegionOperationStatus())
+        .thenReturn(RestoreRedundancyResults.Status.SUCCESS);
+    // isStatusCommand is the fourth argument passed to the function
+    when(mockContext.getArguments()).thenReturn(new Object[] {request, true});
+
+    function.execute(mockContext);
+
+    verify(mockOperation, times(1)).redundancyStatus();
+    verify(mockOperation, times(0)).start();
+  }
+
+  @Test
+  public void executeFunctionReturnsErrorWhenResultStatusIsError() {
+    when(mockResults.getRegionOperationStatus()).thenReturn(RestoreRedundancyResults.Status.ERROR);
+    function.execute(mockContext);
+    verify(resultSender).lastResult(argumentCaptor.capture());
+
+    RestoreRedundancyResults result = argumentCaptor.getValue();
+    assertThat(result.getSuccess()).isFalse();
+    assertThat(result.getStatusMessage()).isEqualTo(message);
+  }
+
+  @Test
+  // The function was able to execute successfully but redundancy was not able to be established for
+  // at least one region
+  public void executeFunctionReturnsOkWhenResultStatusIsFailure() {
+    when(mockResults.getRegionOperationStatus())
+        .thenReturn(RestoreRedundancyResults.Status.FAILURE);
+    function.execute(mockContext);
+    verify(resultSender).lastResult(argumentCaptor.capture());
+
+    SerializableRestoreRedundancyResultsImpl result = argumentCaptor.getValue();
+    verify(result).setSuccess(true);
+    assertThat(result.getRegionOperationStatus())
+        .isEqualTo(RestoreRedundancyResults.Status.FAILURE);
+    assertThat(result).isSameAs(mockResults);
+  }
+
+  @Test
+  public void executeFunctionReturnsOkWhenResultStatusIsSuccess() {
+    when(mockResults.getRegionOperationStatus())
+        .thenReturn(RestoreRedundancyResults.Status.SUCCESS);
+    function.execute(mockContext);
+    verify(resultSender).lastResult(argumentCaptor.capture());
+
+    SerializableRestoreRedundancyResultsImpl result = argumentCaptor.getValue();
+    verify(result).setSuccess(true);
+    assertThat(result.getRegionOperationStatus())
+        .isEqualTo(RestoreRedundancyResults.Status.SUCCESS);
+    assertThat(result).isSameAs(mockResults);
+  }
+
+  @Test
+  public void whenFunctionThrowException() throws Exception {
```

Comment:
This test name could be a little more descriptive, saying what the expected behaviour is given the test conditions, such as "executeFunctionReturnsFailureResultWhenExceptionIsThrownDuringOperation". Also, an exception is never thrown from this method, so the `throws Exception` can be removed from the method signature.